### DISCUSSION TITLE
Resolve App/Form Keyboard Tab Navigation with proper a11y outlines

### DISCUSSION THREAD

**Discussion by manuelwehrli** [directus/directus#4825] [Feature Requests]
When pressing the tab key inside a modal of the admin app, for example editing fields of a collection, the focus does not go linearly from one to the next element, sometimes it seems to get stuck on an invisible element in between (maybe on the hidden dropdown menu to edit the raw value), and on number fields the tab key puts the focus on each of that increment/decrement button inside the field, so it only gets to the next field after pressing three times. Can we have a more standard tab behavior?
Generally, there is no visual feedback where the current focus is on. Could this be improved so we can make use of the default key shortcuts, e.g. space bar to select/deselect checkbox or to open a dropdown menu?

Environment: Directus v9.0.0-rc.53 / Postgres 13.2 / Docker / Firefox 87.0

**Comment by Nitwel** [comment]
Directus right now is not really designed to work with tab navigation. This will probably change in the future because its a very important keypoint in accessability but for now, it's not really a main goal to have that working.

**Comment by benhaynes** [comment]
Correct. To get this right will require a dedicated, app-wide sprint. We are wrapping up bugs and v9 now, but this (and other a11y issues) are high on the list after that.

**Comment by zenithlight** [comment]
Is there a tracking issue for this? I would like to stay updated on the status of this, and I can PR when I have time/if that would be helpful. This feature would be immensely useful for data entry :slightly_smiling_face:

**Reply by benhaynes** [reply]
This is the ticket tracking the issue... you are subscribed and will receive updates.

**Comment by arpadgabor** [comment]
Hello, has there been any progress on this?

At least for me, there are 2 huge pain points here:
- Focus trapping inside modals - focus goes who knows where on the page after too many tabs
- Keyboard navigation of elements in a list box (e.g. Adding translations is a real burden when typing your country name and expecting to have the first result focused to press enter)

At least in the case of focus trapping, Headless UI has a [nice component](https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-vue/src/components/focus-trap/focus-trap.ts) and an interesting way to handle teleports (they place an invisible element before the first focusable element and another one after the last and when you tab into them you are automatically focused back to the first focusable element) and also handle nested focus traps (as is often common in Directus to have multiple nested modals).

**Reply by rijkvanzanten** [reply]
For focus trapping specifically, we can start relying on the html native `<dialog>` element nowadays 😄 https://caniuse.com/dialog

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
